workers/unity/Packages/com.improbable.gdk.mobile/Utility/MobileConnectionFlowInitializer.cs

```diff
@@ -0,0 +1,142 @@
+using System.Linq;
+using Improbable.Gdk.Core;
+using UnityEngine;
+
+namespace Improbable.Gdk.Mobile
+{
+    public class MobileConnectionFlowInitializer : IConnectionFlowInitializer<ReceptionistFlow>,
+        IConnectionFlowInitializer<AlphaLocatorFlow>
+    {
+        private readonly IMobileSettingsProvider[] settingsProviders;
+
+        public MobileConnectionFlowInitializer(params IMobileSettingsProvider[] settingsProviders)
+        {
+            this.settingsProviders = settingsProviders;
+        }
+
+        public ConnectionService GetConnectionService()
+        {
+            return settingsProviders
+                .Select(provider => provider.GetConnectionService())
+                .FilterOption()
+                .FirstOrDefault();
+        }
+
+        public void Initialize(ReceptionistFlow receptionist)
+        {
+#if UNITY_ANDROID
+            // Override for Android emulator.
+
+            if (Application.isMobilePlatform && DeviceInfo.ActiveDeviceType == MobileDeviceType.Virtual)
+            {
+                receptionist.ReceptionistHost = DeviceInfo.AndroidEmulatorDefaultCallbackIp;
+                return;
+            }
+#endif
+            receptionist.ReceptionistHost = settingsProviders
+                .Select(provider => provider.GetReceptionistHostIp())
+                .FilterOption()
+                .FirstOrDefault() ?? RuntimeConfigDefaults.ReceptionistHost;
+        }
+
+        public void Initialize(AlphaLocatorFlow alphaLocator)
+        {
+            alphaLocator.DevAuthToken = settingsProviders
+                .Select(provider => provider.GetDevAuthToken())
+                .FilterOption()
+                .FirstOrDefault() ?? string.Empty;
+        }
+
+        public interface IMobileSettingsProvider
+        {
+            Option<string> GetReceptionistHostIp();
+            Option<string> GetDevAuthToken();
+            Option<ConnectionService> GetConnectionService();
+        }
+
+        public class CommandLineSettingsProvider : IMobileSettingsProvider
+        {
+            public Option<string> GetReceptionistHostIp()
+            {
+                var args = LaunchArguments.GetArguments();
+
+                var hostIp = string.Empty;
+
+                if (args.TryGetCommandLineValue(RuntimeConfigNames.ReceptionistHost, ref hostIp))
```

Comment:
invert ifs in `CommandLineSettingsProvider`

(seems you've done that with `GetConnectionService` in `PlayerPrefsSettingsProvider`)